Runtime/Folder.cs

```diff
@@ -56,6 +57,19 @@ public class Folder : MonoBehaviour
 
         private static Tool lastTool;
         private static Folder toolLock;
+        
+        public static HashSet<int> folders = new HashSet<int>();
+        //public static ReadOnlyCollection<int> folders => new ReadOnlyCollection<int>(_folders.ToList());
+
+        private void Start()
+        {
+            folders.Add(gameObject.GetInstanceID());
+        }
+
+        private void OnDestroy()
+        {
+            folders.Remove(gameObject.GetInstanceID());
+        }
```

Comment:
What specifically is this intended to do?

Reply:
This works as a cache of all the loaded folders. This cache helps to optimize the icon injection procedure as we only need to do know if a given instanceId is contained that HashSet to perform the icon injection.

I decided to have it on the actual Folder class to avoid the Folder class to depend on something else. This could certainly be refactored for better clarity.